### DISCUSSION TITLE
Can't use getInitialProps and getServerSideProps together

### DISCUSSION THREAD

**Discussion by samynaj** [vercel/next.js#18233] [Help]
I am actually improving on the code base of this nextjs project and am faced with the issue of a route protection HOC in the utils folder of the project that utilizes getInitialProps and with the deprecation of getInitialProps in Next 9.5 and introduction of getServerSideProps I have been trying to use getServerSideProps in my pages but I get this error that I can't use getInitialProps and getServerSideProps together. I also know getServerSideProps doesn't work outside of a page. I want to convert the getInitialProps so I can use getServerSideProps in my pages.

**Comment by msreekm** [comment]
I tried using getinitialprops inside my custom document and getServerSideProps in my page and it worked. 
 https://github.com/msreekm/nextjs-material-ui-styled-components-boilerplate

**Reply by ljosberinn** [reply]
getInitialProps in _document is still valid and has nothing to do with the original question asked

**Reply by msreekm** [reply]
okay, I misunderstood the question. ty

**Comment by ljosberinn** [comment]
converting getInitialProps to getServerSideProps is usually trivial, what issues are you seeing?

**Reply by samynaj** [reply]
ok, @ljosberinn I get the error that I can't use getInitialProps and getServerSideProps, this is because the route protection uses getInitialProps and I use it to wrap the page component that also exports a getServerSideProps for serverside rendering. So the only solution I see on site is reverting my pages to exporting getInitialProps, else I cant see a workaround if I insist on using getServerSideProps in my pages.

**Reply by ljosberinn** [reply]
yes, you can only use getInitialProps OR getServerSideProps/getStaticProps. you can convert getInitialProps to getServerSideProps. your getInitialProps is redundant now.
